extensions/ql-vscode/src/upgrades.ts

```diff
@@ -14,79 +18,101 @@ import { upgradesTmpDir } from './run-queries';
 const MAX_UPGRADE_MESSAGE_LINES = 10;
 
 /**
- * Checks whether the given database can be upgraded to the given target DB scheme,
- * and whether the user wants to proceed with the upgrade.
- * Reports errors to both the user and the console.
- * @returns the `UpgradeParams` needed to start the upgrade, if the upgrade is possible and was confirmed by the user, or `undefined` otherwise.
+ * Check that we support non-destructive upgrades.
+ * 
+ * This requires 3 features. The ability to compile an upgrade sequence; The ability to 
+ * run a non-destructive upgrades as a query; the ability to specify a target when 
+ * resolving upgrades.
  */
-async function checkAndConfirmDatabaseUpgrade(
-  qs: qsClient.QueryServerClient,
+export async function hasNondestructiveUpgradeCapabilities(qs: qsClient.QueryServerClient): Promise<boolean> {
+  // TODO change to actual version when known
+  // Note it is probably something 2.4.something
+  return semver.gte(await qs.cliServer.getVersion(), '2.4.1');
+}
+
+
+/**
+ * Compile a database upgrade sequence.
+ * Callers must check that this is valid with the current queryserver first.
```

Comment:
Any reason not to do this check inside this function? The version info is retrieved once and cached, so it is usually a fast operation.